plugins/modules/route53_health_check.py

```diff
@@ -268,6 +301,20 @@ def find_health_check(ip_addr, fqdn, hc_type, request_interval, port):
             return None
 
 
+def get_existing_checks_with_name():
+    results = _list_health_checks()
+    health_checks_with_name = {}
+    while True:
+        for check in results.get('HealthChecks'):
+            if 'Name' in describe_health_check(check['Id'])['tags']:
+                check_name = describe_health_check(check['Id'])['tags']['Name']
+                health_checks_with_name[check_name] = check
+        if results.get('IsTruncated', False):
+            results = _list_health_checks(Marker=results.get('NextMarker'))
+        else:
+            return health_checks_with_name
```

Comment:
```suggestion
    while True:
        for check in results.get('HealthChecks'):
            if 'Name' in describe_health_check(check['Id'])['tags']:
                check_name = describe_health_check(check['Id'])['tags']['Name']
                health_checks_with_name[check_name] = check
        if results.get('IsTruncated', False):
            results = _list_health_checks(Marker=results.get('NextMarker'))
        else:
            return health_checks_with_name
```

hm we should try avoid `while True:`.  
You can do something like [that](https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/wafv2_ip_set_info.py#L83-L98), where the function calls itself, based in the return value of `NextMarker`.  
but on the other hand do we have multiple `while True:` constructs already in c.a